packages/sdks/react/src/lib/useProfile.ts

```diff
@@ -2,39 +2,69 @@ import { useState, useEffect, useRef } from 'react';
 import { isEqual } from 'radash';
-
 import { useNinetailed } from './useNinetailed';
 
+/**
+ * Custom hook that provides access to the Ninetailed profile state
+ * with the 'experiences' property removed to prevent unnecessary re-renders.
+ *
+ * This hook handles profile state changes efficiently by:
+ * 1. Only updating state when actual changes occur
+ * 2. Removing the large 'experiences' object from the state
+ * 3. Properly cleaning up subscriptions when components unmount
+ *
+ * @returns The profile state without the 'experiences' property
+ */
 export const useProfile = () => {
+  // Get the Ninetailed instance
   const ninetailed = useNinetailed();
-  const [profileState, setProfileState] = useState<ProfileState>(
-    ninetailed.profileState
-  );
-  const profileStateRef = useRef({});
-
-  /**
-   * This effect compares the old and new profile state before updating it.
-   * We use a ref to avoid an infinite loop which can happen when an empty profile state was updated with no changes.
-   * This behaviour occurred as the validation handling on the error property was not set properly in the "CreateProfile" and "UpdateProfile" endpoint types.
-   * Furthermore, it was also observed, that it "only" occurred when the preview plugin was used in parallel.
-   */
+
+  // Extract initial state without experiences
+  const { experiences, ...profileStateWithoutExperiences } =
+    ninetailed.profileState;
+
+  // State to hold the stripped profile state
+  const [strippedProfileState, setStrippedProfileState] = useState<
+    Omit<ProfileState, 'experiences'> & { loading: boolean }
+  >({
+    ...profileStateWithoutExperiences,
+    loading: ninetailed.profileState.status === 'loading',
+  });
+
+  // Reference to track the previous profile state for comparison
+  const profileStateRef = useRef(ninetailed.profileState);
+
   useEffect(() => {
-    ninetailed.onProfileChange((profileState) => {
-      if (isEqual(profileState, profileStateRef.current)) {
-        logger.debug('Profile State Did Not Change', profileState);
+    // Subscribe to profile changes
+    const unsubscribe = ninetailed.onProfileChange((changedProfileState) => {
+      // Skip update if the profile hasn't actually changed
+      // Here we compare the entire profile including experiences and changes
+      if (isEqual(changedProfileState, profileStateRef.current)) {
+        logger.debug('Profile State Did Not Change', changedProfileState);
         return;
-      } else {
-        setProfileState(profileState);
-        profileStateRef.current = profileState;
-        logger.debug('Profile State Changed', profileState);
       }
+
+      // Update the ref and log the change
+      profileStateRef.current = changedProfileState;
+      logger.debug('Profile State Changed', changedProfileState);
+
+      // Extract everything except experiences and update state
+      const { experiences, ...profileStateWithoutExperiences } =
```

Comment:
This could move into a helper function, as you do the stripping and adding the loading flag twice